core/asset/service_test.go

```diff
@@ -195,7 +196,7 @@ func TestService_UpsertAsset(t *testing.T) {
 			Asset:       sampleAsset,
 			Setup: func(ctx context.Context, ar *mocks.AssetRepository, dr *mocks.DiscoveryRepository, lr *mocks.LineageRepository) {
 				ar.EXPECT().Upsert(ctx, sampleAsset).Return(sampleAsset.ID, nil)
-				dr.EXPECT().Upsert(ctx, *sampleAsset).Return(errors.New("unknown error"))
+				dr.EXPECT().Upsert(ctx, mock.AnythingOfType("asset.Asset")).Return(errors.New("unknown error"))
```

Comment:
because in implementation, before Upsert the asset, it updates the refreshed_at field to be current time at that time, so the mock need like that to cover that. Will got error if still *sampleAsset (pointer) even if I update the refreshed_at in sampleAsset, because it is different value in refreshed_at in miliseconds